### DISCUSSION TITLE
Script to Berry of upgraded chip esp8266 to esp32c3

### DISCUSSION THREAD

**Discussion by kroon040** [arendst/Tasmota#20841] [Q&A]
Hi,

I upgrade a chip from esp8266 to an esp32c3. I had scripting enabled but I want to have a more clean bin file that I can use without compiling additional options. I have no knowledge of berry and tried alslo chatgpt but he can;t convert my script to berry. Can somebody do this? Also I want to see the values in de webinterface and use it with mqtt. I compiled this with scripting enabled. I hope that the default image with berry this is not needed anymore

#ifndef USE_SCRIPT
#define USE_SCRIPT
  #define USE_SCRIPT_WEB_DISPLAY
  #define SUPPORT_MQTT_EVENT                  
#endif

Below the script

```
>D
tin=0
tout1=0
hin=0
hout1=0
temp=""
>E
tin=TuyaReceived#DpType2Id1/10
hin=TuyaReceived#DpType2Id2
hout1=TuyaReceived#DpType2Id103
temp=TuyaReceived#102#DpIdData
if ins(temp "FFFF")==-1 { 
tout1=hd(sb(temp 4 4))/10
}else{
tout1=hd(sb(temp 4 4))-65536/10
}
>T
=>publish bresser/schuur/temp %1tin%
=>publish bresser/schuur/hum %0hin%
=>publish bresser/buiten/temp %1tout1%
=>publish bresser/buiten/hum %0hout1%
>W
TempIndoor{m} %1tin% °C
HumIndoor{m} %0hin% %%
TempOutdoor{m} %1tout1% °C
HumOutdoor{m} %0hout1% %%
```

**Comment by sfromis** [comment]
Here's an example illustrating how you can create a driver based on JSON payloads, and showing data on the front page:
```
class SampleDriver
  var uptime, heap
  def init()
    var id = 'SampleDriver'
    tasmota.remove_driver(global.SampleDriver_instance)
    tasmota.remove_rule('UptimeSec', id)
    tasmota.add_rule('UptimeSec', /values, trigger, payload -> self.processState(values, trigger, payload), id)
    tasmota.add_driver(global.SampleDriver_instance := self)
  end
  def processState(value, trigger, payload)
    self.uptime = value
    self.heap = payload.find('Heap',-1)
  end
  def web_sensor()
    if self.uptime
      tasmota.web_send_decimal(format('{s}Uptime{m}%i{e}{s}Heap{m}%i{e}', self.uptime, self.heap))
    end
  end
  def json_append()
    if self.uptime
      tasmota.response_append(f',"MySensor":{{"Uptime":{self.uptime},"Heap":{self.heap}}}')
      #import json
      #var sensorobject = {'MySensor':{'Uptime':self.uptime,'Heap':self.heap}}
      #tasmota.response_append(',' + json.dump(sensorobject))
    end
  end
end
sd=SampleDriver()
```
In `json_append` the commented lines illustrate another way of creating the sensor payload instead of a string. You can replace the method by something like:
```
  def json_append()
    if self.uptime
      import mqtt
      mqtt.publish('bresser/schuur/uptime', str(self.uptime))
      mqtt.publish('bresser/schuur/heap', str(self.heap))
    end
  end
```

Note that the specific example is somewhat silly, and only reacts when you use the `State` command. The point is to illustrate the mechanism, not to be useful.

The point about having `global.SampleDriver_instance` is that the driver can replace itself if you input a new iteration, without the old driver lingering on.

**Reply by kroon040** [reply]
Thnx for your answer. I have no experience with programming , I don’t know how to do it

**Reply by sfromis** [reply]
In the example above, you could get some way by just changing the code to use the fields you're after, while keeping the same structure. And if you created the script you posted, you do have a bit of experience with programming, just a very different language.

**Comment by kroon040** [comment]
I have my old script from the forums right here. (tasmota with bresser weatherstation), so no knowledge...

**Comment by kroon040** [comment]
I have now this , but this is not enoughI think, to have it in the web interface and published with mqtt


```
tin = 0
tout1 = 0
hin = 0
hout1 = 0
temp = ""

tin = TuyaReceived#DpType2Id1 / 10
hin = TuyaReceived#DpType2Id2
hout1 = TuyaReceived#DpType2Id103
temp = TuyaReceived#102#DpIdData

if ins(temp, "FFFF") == -1 { 
    tout1 = hd(sb(temp, 4, 4)) / 10
} else {
    tout1 = hd(sb(temp, 4, 4)) - 65536 / 10
}

publish "bresser/schuur/temp" %1tin%
publish "bresser/schuur/hum" %0hin%
publish "bresser/buiten/temp" %1tout1%
publish "bresser/buiten/hum" %0hout1%

"TempIndoor{m}" %1tin% "°C"
"HumIndoor{m}" %0hin% "%%"
"TempOutdoor{m}" %1tout1% "°C"
"HumOutdoor{m}" %0hout1% "%%"

```
